src/cupbearer/tasks/tiny_natural_mechanisms.py

```diff
@@ -13,6 +14,54 @@
 from .task import Task
 
 
+def decode_and_encode(tokens, task_model, model):
```

Comment:
Nit: maybe `decoder_model` and `encoder_model` (or `new_model` or something) to make it clearer what the arguments are?